src/write-results-to-db.js

```diff
@@ -0,0 +1,110 @@
+const knex = require("knex")
+const moment = require("moment-timezone")
+const config = require("./config")
+
+const writeResultsToDatabase = (results, { realtime } = {}) => {
+  const db = knex({ client: "pg", connection: config.postgres })
+
+  if (realtime) {
+    return _writeRealtimeResults({ db, results }).then(() => db.destroy())
+  } else if (results.query.dimensions.match(/ga:date/)) {
+    return _writeRegularResults({ db, results }).then(() => db.destroy())
+  } else {
+    return Promise.resolve()
+  }
+}
+
+const _dataForDataPoint = (dataPoint, { realtime } = {}) => {
+  const data = Object.assign({}, dataPoint)
+  let dateTime
+  if (realtime) {
+    dateTime = (new Date()).toISOString()
+  } else {
+    dateTime = _dateTimeForDataPoint(dataPoint)
+  }
+  delete data.date
+  delete data.hour
+
+  return {
+    date_time: dateTime,
+    data: data,
+  }
+}
+
+const _dateTimeForDataPoint = (dataPoint) => {
+  let dateString = dataPoint.date
+  if (dataPoint.hour) {
+    dateString = `${dateString}T${dataPoint.hour}:00:00`
+  } else {
+    dateString = `${dateString}T00:00:00`
+  }
+  if (!isNaN(Date.parse(dateString))) {
+    return moment.tz(dateString, config.timezone).toISOString()
+  }
+}
+
+const _queryForExistingRow = ({ db, row }) => {
+  query = db("analytics_data")
```

Comment:
And since it's used in multiple places, I think it should be stored in a central var somewhere regardless.